addons/website_forum/controllers/main.py

```diff
@@ -164,20 +164,26 @@ def tag_read(self, query='', limit=25, **post):
 
     @http.route(['/forum/<model("forum.forum"):forum>/tag', '/forum/<model("forum.forum"):forum>/tag/<string:tag_char>'], type='http', auth="public", website=True, sitemap=False)
     def tags(self, forum, tag_char=None, **post):
+        Tag = request.env['forum.tag']
         # build the list of tag first char, with their value as tag_char param Ex : [('All', 'all'), ('C', 'c'), ('G', 'g'), ('Z', z)]
         first_char_tag = forum.get_tags_first_char()
         first_char_list = [(t, t.lower()) for t in first_char_tag if t.isalnum()]
         first_char_list.insert(0, (_('All'), 'all'))
 
+        domain = [('forum_id', '=', forum.id), ('posts_count', '>', 0)]
         active_char_tag = tag_char and tag_char.lower() or 'all'
 
+        # perf issue, fixed in master, to remove in master
+        if active_char_tag == 'all' and Tag.search_count(domain) > 200:
+            redirect_url = "/forum/%s/tag/%s" % (slug(forum), first_char_list[1][1])
+            return werkzeug.utils.redirect(redirect_url, 301)
```

Comment:
I would use 302 in this case, but @JKE-be knows better than me.

Reply:
I am open to both 301 or 302.
I thought about it, but in this particular case I don't know if there is a better one.
That would be a 302 of multiple months, maybe years. Seems like there is some article about that suggesting it would be too long for a 302 but didn't investigate much more than that.

Reply:
302 is perfect is better, and google is smart enough to understand that it is for long term, but at least your browser will not cache redirect for next version.

> Google recognizes that many people use 302 when they really mean 301. Fortunately, Google isn’t bound by any law to take people literally. For the sake of producing the best possible search results, Google can and should look at 302s and figure out if the webmaster really means 302, or if it’s run-of-the-mill confusion and they really mean 301.

src https://www.hochmanconsultants.com/301-vs-302-redirect/

Reply:
I don't think there is a better or not for this case.

From the same article you quote:
> **Whether Google actually handles 302s properly is an open question**. If a 302 is used instead of a 301, search engines might continue to index the old URL, and disregard the new one as a duplicate. **Link popularity might be divided between the two urls, hurting search rankings**. Search engines might figure out how to handle the 302, **or they might not**. Google spokespeople have said that they will treat a 302 as a 301 **if they think the webmaster has made an error, but why take a chance, and what about other search engines**?

Also, other articles seems to tend to use 301 if the redirect will be there for a long period of time, which is the case here, no one will be migrated to 14.0 before a year at least, and people won't migrate directly so they will be using the redirection for **years**.

> When is a 301 Redirect Appropriate?
[...]
Your previous page will be replaced and the new page will be shown in your search engine results with a 301 redirect. **This is appropriate when you know the URL will be at the new page for a year or longer.** 

src https://www.seoclarity.net/resources/knowledgebase/use-301-redirect-vs-302-redirect-15683/

> What Is a 301 Redirect?
A 301 redirect sends the message to search engines that a website or page has been moved permanently.
**Permanent means around a year or longer.** After a year, check to see if people are still being redirected to your site.

src https://www.searchenginejournal.com/301-vs-302-redirects-seo/299843/#close

As I said, I don't really mind using a 301 or 302 here as there won't be much impact. 301 seems a little better but I will change it if you both prefer 302 as I am fine with it too.